lib/matplotlib/widgets.py

```diff
@@ -2013,8 +2016,8 @@ def on_select(min: float, max: float) -> Any
         The direction along which to draw the span selector.
 
     minspan : float, default: 0
-        If selection is less than or equal to *minspan*, do not call
-        *onselect*.
+        If selection is less than or equal to *minspan* and larger than 0,
+        do not call *onselect*.
```

Comment:
Is this change still correct after the last changes? (I lost track...)
ditto below.

Reply:
No, not quite, it still calls it when the selection "has been completed" and becomes less than `minspan`...

Reply:
I guess, the simplest is not to call it when the span width is less than `minspan` in any case?

Reply:
I *think* it should be always called, *except* if selection_completed was False and the current selection is also less than minspan? (effectively going from no selection to still no selection)

Reply:
Yes, this makes more sense to cover the following behaviour:
- when the selection is new and less than minspan*, don't call onselect, because the selection is not "valid/completed"
- when the selection is already existing and made less than minspan*, call onselect, so that the callback can know that the selection is not "valid/completed" anymore/has been removed (the callback still need to be responsible about this) and do what it needs to do!

This behaviour is implemented in 9e510a5 and the docstrings have been updated accordingly.